### PR TITLE
fix: deduplicate signatures in SignatureSet.Add

### LINKED ISSUE #6714: Add tests for adding a signature from the same validator twice
> [nit]
> to avoid mistakes, we could check whether this signature was already added. If so, return an error or just nil. Otherwise, we risk increasing the voting power by mistake if we call this method twice on the same validator/signature 

 _Originally posted by @rach-id in [#6706](https://github.com/celestiaorg/celestia-app/pull/6706/changes/BASE..a4972124c672ea7f43be431c27e1ade23767ff3f#r2883503069)_

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://linear.app/celestia/issue/DA-1153/add-tests-for-adding-a-signature-from-the-same-validator-twice
Closes #6714

## Summary

- Check for duplicate validator signatures in `SignatureSet.Add()` to prevent double-counting voting power
- Add tests for duplicate signature handling (sequential and concurrent)

## Test plan

- [x] `go test -v -run TestSignatureSet ./fibre/validator/` passes (including new `DuplicateSignatureNoOp` and `DuplicateSignatureConcurrent` subtests)
- [x] `go test -race ./fibre/validator/` passes
- [x] `golangci-lint run ./fibre/validator/...` reports 0 issues

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

<a href="https://app.devin.ai/review/celestiaorg/celestia-app/pull/6828" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open with Devin
  </picture>
</a>

### CODE REVIEW

File: fibre/validator/signature_set_test.go [hunk 1]
```diff
@@ -174,6 +174,57 @@ func TestSignatureSet(t *testing.T) {
...
+		require.Error(t, err)
+
+		var sigErr *validator.NotEnoughSignaturesError
+		require.ErrorAs(t, err, &sigErr)
+		require.Equal(t, int64(10), sigErr.CollectedPower)
+		require.Equal(t, int64(15), sigErr.RequiredPower)
+		require.Nil(t, sigs)
+	})
+
+	t.Run("DuplicateSignatureConcurrent", func(t *testing.T) {
```
Comment (Wondertan) [inline fibre/validator/signature_set_test.go:203]: This is a rather useless test tho
